pennylane/pulse/parametrized_hamiltonian.py

```diff
@@ -240,7 +240,18 @@ def __call__(self, params, t):
         return self.H_fixed() + self.H_parametrized(params, t)
 
     def __repr__(self):
-        return f"ParametrizedHamiltonian: terms={qml.math.shape(self.coeffs)[0]}"
+        terms = []
+        for i, (coeff, op) in enumerate(
+            zip(
+                self.coeffs_fixed + self.coeffs_parametrized, self.ops_fixed + self.ops_parametrized
+            )
+        ):
```

Comment:
I think we should separate iterating over `.._fixed` and `.._parametrized` as the value of `i` is offset by the number of fixed terms

Reply:
Yes WIll do it , I will split them into 2 different iteration like this 
```
        for coeff, op in zip(self.coeffs_fixed, self.ops_fixed):
            term = f"{coeff} * {op}"
            terms.append(term)

        for i, (coeff, op) in enumerate(zip(self.coeffs_parametrized, self.ops_parametrized)):
            if callable(coeff) and hasattr(coeff, "__name__")
                term = f"{coeff.__name__}(params_{i}, t) * {op}"
            else:
                term = f"{coeff} * {op}"
            terms.append(term)
            ```